howdoi/howdoi.py

```diff
@@ -148,12 +165,23 @@ def _extract_links(html, search_engine):
 def _get_search_url(search_engine):
     return SEARCH_URLS.get(search_engine, SEARCH_URLS['google'])
 
+def _detect_block(page):
+    for indicator in BLOCK_INDICATORS:
+        if page.find(indicator) != -1:
+            return True
+
+    return False
 
 def _get_links(query):
     search_engine = os.getenv('HOWDOI_SEARCH_ENGINE', 'google')
     search_url = _get_search_url(search_engine)
 
     result = _get_result(search_url.format(URL, url_quote(query)))
+    if _detect_block(result):
+        print('[ERROR] Encountered a temporary block due to automated request '
+                'detection. Please wait a few minutes for it to clear or change search engine')
+        raise BlockError("Temporary block by search engine")
```

Comment:
What about using a standard exception like https://docs.python.org/3/library/exceptions.html#RuntimeError

Reply:
Was considered good practice on other py projects I contributed to. I happen to like it and it makes for cleaner code if we end up handling the exception upstream instead of just oopsing with a tb. Your call, I don't insist.

Reply:
OK let's keep the BlockError, but it should inherit from RuntimeError, yes?

Reply:
If you're asking me, no - RuntimeError is for people too lazy do define their own exceptions properly, says nothing and gives no benefit. I have exactly 0 problem just doing what you ask though :)

Reply:
I'm suggesting the following:

```
class BlockError(RuntimeError):
    pass
```

Here, we get the advantages of a custom error, but we're more explicit about the type of Exception we're handling – one that happens at runtime and thus out of the hands of the developer.

Reply:
I understood what you were suggesting. All exceptions defined in running code are by definition runtime exceptions. I consider inheriting from built-in exceptions unidiomatic (unless it's a clear sub-type) and haven't seen it in Twisted or any other large projects I'm familiar with - they all have their own exception hierarchies. With that said, it's not important and deffo your call.

d3e69832005107745342e04317597ff2fe0aa639